loopin.go

```diff
@@ -690,8 +690,9 @@ func (s *loopInSwap) publishTimeoutTx(ctx context.Context,
 	return nil
 }
 
-// persistState updates the swap state and sends out an update notification.
-func (s *loopInSwap) persistState(ctx context.Context) error {
+// persistAndAnnounceState updates the swap state on disk and sends out an
+// update notification.
+func (s *loopInSwap) persistAndAnnounceState(ctx context.Context) error {
```

Comment:
Wouldn't it be better if we would just have `setState(announce bool)` which would set `lastUpdate`, persist and announce (when requested).

Reply:
We don't always persist immediately if we set state. Every call to the db will result in a new update row being added.

Reply:
Indeed, we append a new state with a sequence id. Should state maybe have a "dirty" flag so that we only serialize when the value changed? Would be nice to have one update function which works in every occasion.

Reply:
Then we should still get more updates than needed? Every change will immediately result in a db update, even if it is not necessary to persist the change yet.

I don't really want to change that mechanism in this PR. It will affect all swap state db actions and possibly also what happens on resume.

Reply:
SGTM